src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
+        if (head == currentNode) {
+            head = node;
+        } else {
+            currentNode.prev.next = node;
+        }
+        node.prev = currentNode.prev;
+        node.next = currentNode;
+        currentNode.prev = node;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = findByIndex(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return checkNode(findByIndex(index)).item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        return checkNode(findByObject(object)) != null;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public void isIndexExist(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " dosen't exist");
+        }
+    }
+
+    public Node<T> findByIndex(int index) {
+        isIndexExist(index);
+        if (head == null) {
+            return null;
+        }
+        Node<T> currentNode;
+        if (index > (size / 2)) {
+            currentNode = tail;
+            for (int i = size - 1; i != index; i--) {
+                currentNode = currentNode.prev;
+            }
+        } else {
+            currentNode = head;
+            for (int i = 0; i != index; i++) {
+                currentNode = currentNode.next;
+            }
+        }
+        return currentNode;
+    }
+
+    private Node<T> findByObject(T object) {
+        Node<T> element = head;
+        do {
+            T value = element.item;
+            if (value == object || value != null && value.equals(object)) {
+                return element;
+            }
+            element = element.next;
+        } while (element.next != null);
+        return null;
+    }
+
+    private Node<T> checkNode(Node<T> node) {
+        if (node == null) {
+            return null;
+        }
+        if (node.prev == null) {
+            head = node.next;
+        } else {
+            node.prev.next = node.next;
+        }
+        if (node.next == null) {
+            tail = node.prev;
+        } else {
+            node.next.prev = node.prev;
+        }
+        size--;
+        return node;
+    }
+
+    private static class Node<T> {
+
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(T value) {
+            this.item = value;
+        }
```

Comment:
what about setting the next and prev value in the constructor too?  You will be able to pass it in constructor whether it's known, and avoid writing it after object creation